### DISCUSSION TITLE
Issue regarding errorFilter.ts error handling.

### DISCUSSION THREAD

**Discussion by kawadker** [TypedProject/ts-express-decorators#2896] [General]
Hi, 
as we can see in the below code that if we give the custom error that " data not found " and if it is production env it will throw error "**InternalServerError**" but in dev env it is working fine is there any specific reason that for prod env it is changing the error forcefully itself 

`    response
      .setHeaders(this.getHeaders(error))
      .status(err.status)
      .contentType("application/json")
      .body(env === Env.PROD ? "InternalServerError" : err);`
      
      This makes production break ideally behaviour should be same in dev and production. We are using tag v7.18.4 
      
      Mostly it happening when we are throwing any error from code with the help of exeption from "ts-httpexceptions" module, but we are throwing error with the extends @tsed/exceptions it works  fine and not throwing any error for prod env as well.

**Comment by SavvasLalitKawadker** [comment]


**Reply by Romakita** [reply]
It’s to avoid security issue.

**Reply by Romakita** [reply]
But this message is only sent when the status code is a 500. So if you emit a 500 for functionnal error, you have problably a design code problem ;)
It should be a 404 error at least

**Reply by Romakita** [reply]
I see a mention ts-httpexceptions. This module isn’t supported since a long time.

**Reply by ckgupta** [reply]
Hi @Romakita thanks for the suggestion, we updated tsed version recently probably we should update to @tsed/exceptions. 

However, we were not able to understand this specific implementation where in InternalServerError is being thrown only if the env is `Env.PROD`   https://github.com/tsedio/tsed/blob/eaadba550948134c58f2f731d4b45a254fa3cbcc/packages/platform/platform-exceptions/src/components/ErrorFilter.ts#L20

Do you think this was intentional? As I understand it, this should be consistent across all environments.

**Reply by Romakita** [reply]
I already answered you it’s for security issue. ErrorFilter is called only for unqualified base Error, that will expose your infrastructure to any hackers. This is the last filter called if any other error filter doesn’t match the class.
Before ErrorFilter, you have the ExceptionFilter, that transform all Exception classes to a json payload.
Because you use an very old exception library, it’s the ErrorFilter that catch your error and not the ExceptionFilter.

https://github.com/tsedio/tsed/blob/eaadba550948134c58f2f731d4b45a254fa3cbcc/packages/platform/platform-exceptions/src/components/ExceptionFilter.ts#L20

Update your code and it’ll works!

**Reply by ckgupta** [reply]
sure thanks for the clarification @Romakita , it works!
